spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/KullbackLeiblerDivergenceCriterion.scala

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2016 The BigDL Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.intel.analytics.bigdl.nn
+
+import com.intel.analytics.bigdl.nn.abstractnn.TensorCriterion
+import com.intel.analytics.bigdl.tensor.{DenseTensorApply, Tensor, TensorFunc2}
+import com.intel.analytics.bigdl.tensor.TensorNumericMath.TensorNumeric
+
+import scala.reflect.ClassTag
+
+/**
+ * This method is same as `kullback_leibler_divergence` loss in keras.
+ * It calculates:
```

Comment:
Put the formula to Python layer as well?

Reply:
Thanks @wzhongyuan  I will make a pass for the code tomorrow and will let you know once this is ready for review.